compiler/rustc_mir_transform/src/check_unsafety.rs

```diff
@@ -368,7 +368,10 @@ impl<'tcx> UnsafetyChecker<'_, 'tcx> {
         if !callee_features.iter().all(|feature| self_features.contains(feature)) {
             self.require_unsafe(
                 UnsafetyViolationKind::General,
-                UnsafetyViolationDetails::CallToFunctionWith,
+                UnsafetyViolationDetails::CallToFunctionWith {
+                    func_did: func_did,
```

Comment:
Since all you're using this for is for extracting the features that are missing again, you could also collect those features into a `Vec` directly above and just check if the vec is not empty. Then you can store the `Vec` in `CallToFunctionWith` and don't need any `DefIds`.

Reply:
@oli-obk, there's one problem, if I try to store the `Vec` as you suggest I'll get the following error:
```
41  |     CallToFunctionWith { missing_features: Vec<Symbol> },
    |                          ----------------------------- this field does not implement `std::marker::Copy`
```

And it's explained here: https://users.rust-lang.org/t/cant-derive-copy-because-of-string/18665

If you have any idea how can I solve this problem, can you please explain it to me?

For now, I'll continue with the implementation that I had and try to fix my code according to the repo updates.

Reply:
That error is often encountered during development, and you will probably in some place have to clone or use it via a reference instead of dereferencing it somewhere. It's hard to tell without knowing the full error. Maybe copy paste the full error?

Also, in general the better place for such questions is the [help zulip channel](https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp).

Reply:
Thank you @est31, I'll ask about it in the Zulip

Reply:
Now I have it working, but a lot of code had to be changed, I don't know if it's really worth it, but I will push the changes